src/PartnerConfiguration.ts

```diff
@@ -0,0 +1,124 @@
+  'function getMinLength() external view returns (uint256)',
+  'function getMaxLength() external view returns (uint256)',
+  'function getUnicodeSupport() external view returns (bool)',
+  'function getMinDuration() external view returns (uint256)',
+  'function getMaxDuration() external view returns (uint256)',
+  'function getFeePercentage() external view returns (uint256)',
+  'function getDiscount() external view returns (uint256)',
+  'function getMinCommitmentAge() external view returns (uint256)',
+  'function getPrice(string memory name, uint256 expires, uint256 duration) external view returns (uint256)',
+  'function setMinCommitmentAge(uint256 minCommitmentAge) external',
+  'function setDiscount(uint256 discount) external',
+  'function setFeePercentage(uint256 feePercentage) external',
+  'function setMaxDuration(uint256 maxDuration) external',
+  'function setUnicodeSupport(bool flag) external',
+  'function setMaxLength(uint256 maxLength) external',
+  'function setMinDuration(uint256 minDuration) external',
+  'function setMinLength(uint256 minLength) external',
+  'function validateName(string memory name, uint256 duration) external view'
+]
+
+export class PartnerConfiguration {
+  private readonly partnerConfiguration: Contract
+
+  constructor (private readonly partnerConfigurationAddress: string, private readonly signer?: Signer, private readonly provider: string = 'http://localhost:8545') {
+    this.partnerConfiguration = new Contract(partnerConfigurationAddress, partnerConfigurationInterface, new providers.JsonRpcProvider(provider))
+    this.signer = signer
+  }
+
+  getPartnerConfiguration (): Contract { return this.partnerConfiguration }
+  getProvider (): string { return this.provider }
+
+  async getMinLength (): Promise<BigNumber> {
+    return this.partnerConfiguration.getMinLength()
+  }
+
+  getMaxLength (): Promise<BigNumber> {
+    return this.partnerConfiguration.getMaxLength()
+  }
+
+  getUnicodeSupport (): Promise<boolean> {
+    return this.partnerConfiguration.getUnicodeSupport()
+  }
+
+  getMinDuration (): Promise<BigNumber> {
+    return this.partnerConfiguration.getMinDuration()
+  }
+
+  getMaxDuration (): Promise<BigNumber> {
+    return this.partnerConfiguration.getMaxDuration()
+  }
+
+  getMinCommitmentAge (): Promise<BigNumber> {
+    return this.partnerConfiguration.getMinCommitmentAge()
+  }
+
+  getFeePercentage (): Promise<BigNumber> {
+    return this.partnerConfiguration.getFeePercentage()
+  }
+
+  getDiscount (): Promise<BigNumber> {
+    return this.partnerConfiguration.getDiscount()
+  }
+
+  getPrice (name: string, expires: BigNumber, duration: BigNumber): Promise<BigNumber> {
+    return this.partnerConfiguration.getPrice(name, expires, duration)
+  }
+
+  validateName (name: string, duration: BigNumber): Promise<void> {
+    return this.partnerConfiguration.validateName(name, duration)
+  }
+
+  async setDiscount (discount: BigNumber, signer?: Signer): Promise<void> {
```

Comment:
even tho the current implementation doesnt have it, can we use the same documentation style we use in the Solidity code? aiming to automate the docs generation and intellisense when using the functions.